sql-db/hibernate/src/main/java/io/quarkus/qe/hibernate/items/Item.java

```diff
@@ -15,7 +17,7 @@ public class Item {
     public String note;
 
     @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "customerId")
+    @JoinColumn(name = "customerid")
```

Comment:
I don't really like this. To me, it is needlessly complicated to decipher what is going on here.

Here is what I've managed to understand, let's see if I'm right:
- The way it was done before could be considered a showcase of the fact that the column name is case-insensitive in both column definition and in the import script.
- Now, because the quoting strategy is `all-except-column-definitions` and the column is explicitly quoted here, the case in the column definition matters. However, the import script is still interpreted in case-insensitive mode, because the column name is not quoted over there.

So, it is imo doing 3 things at once:
- testing the quoting strategy
- using explicit quoting
- relying on case-insensitive interpretation in the import script

All of which without a proper test, that would give it a bit of context.

I would appreciate a form that is more comprehensible. I'm ok with keeping the code, but let's add some comments both here and in the `import.sql`, which would explain the situation.

Reply:
no, mmnt, wrong branch, I'll re-check

Reply:
> I don't really like this. To me, it is needlessly complicated to decipher what is going on here.

The point here is that we can't create table with "customerId" and refer to it with "customerId" when case sensitivity is important. It is legitimate case for you can see from linked issue there are users who need to use it.

> The way it was done before could be considered a showcase of the fact that the column name is case-insensitive in both column definition and in the import script.

If supported Quarkus extension provides option that users (and intentional customers) can use, we need to know it works (in theory, we can't cover everything). Every other module that tests Hibernate use case insensitivity, so I need to add it somewhere. It strikes me as interesting issue because when our customers are going to migrate their tables, it can possibly happen that their legacy tables will now be called with reserved / key words.

> All of which without a proper test, that would give it a bit of context.

It fails without changes which verifies the issue.


My conclusion: What I can do is to escape column in insert statement myself and keep customerId/customerid as it was. Also I can add comments as you suggested.

Thanks.